pkg/customresourcestate/registry_factory.go

```diff
@@ -152,8 +152,8 @@ type compiledMetric interface {
 
 // newCompiledMetric returns a compiledMetric depending on the given metric type.
 func newCompiledMetric(m Metric) (compiledMetric, error) {
-	switch m.Type {
-	case MetricTypeGauge:
+	switch metric.Type(strings.ToLower(string(m.Type))) {
```

Comment:
do we really need to convert to lowercase here since the config should've already been overriden?